src/nspawn/nspawn.c

```diff
@@ -3974,6 +3974,20 @@ static int outer_child(
         if (r < 0)
                 return log_error_errno(r, "Failed to move root directory: %m");
 
+        if (arg_userns_mode != USER_NAMESPACE_NO) {
+                /* In order to mount procfs and sysfs in an unprivileged container the kernel
+                 * requires that a fully visible instance is already present in the target mount
+                 * namespace. Mount one here so the inner child can mount its own instances. Later
+                 * we umount the temporary instances created here before we actually exec the
+                 * payload. Since the rootfs is shared the umount will propagate into the container.
+                 * Note, the inner child wouldn't be able to unmount the instances on its own since
+                 * it doesn't own the originating mount namespace. IOW, the outer child needs to do
```

Comment:
Technically, this is imprecise. The issue isn't that the inner child doesn't own the originating mount namespace. It's that because at the time the mounts were copied the inner child didn't own the originating mount namespace which caused the to become locked mounts which cannot be unmounted. The can only be unmounted due to propagation events from the originating mount namespace.

Reply:
An alternative solution ofc is to just premount procfs and sysfs here but that will have the consequence that once again they won't be owned by the payload which is undesirable in my design view. It means that the workload has procfs and sysfs instances which it seemingly doesn't own and umount /proc and umount /sys will fail with EINVAL. We should really avoid that.

Reply:
In addition, it won't work because we need the pid namespace of the target to be active.